openhtf/io/output/console_summary.py

```diff
@@ -0,0 +1,107 @@
+"""Module to display test summary on console."""
+
+import os
+import sys
+import __main__ as main
+
+from openhtf import util
+from openhtf.io import test_record
+from openhtf.util import measurements as measurement
+
+
+class ConsoleSummary():
+  """Print test results with failure info on console. """
+
+  # pylint: disable=invalid-name
+  def __init__(self):
+    if os.name == 'posix':    #Linux and Mac
+      self.RED = '\033[91m'
+      self.GREEN = '\033[92m'
+      self.ORANGE = '\033[93m'
+      self.RESET = '\033[0m'
+      self.BOLD = '\033[1m'
+    else:
+      self.RED = ""
+      self.GREEN = ""
+      self.ORANGE = ""
+      self.RESET = ""
+      self.BOLD = ""
+
+    self.color_table = {
+        test_record.Outcome.PASS:self.GREEN, 
+        test_record.Outcome.FAIL:self.RED, 
+        test_record.Outcome.ERROR:self.ORANGE, 
+        test_record.Outcome.TIMEOUT:self.ORANGE, 
+    }
+  # pylint: enable=invalid-name
+
+  def __call__(self, record):
+    indent2 = ' '*2
+    indent4 = ' '*4
+    color = self.color_table[record.outcome]
+    output_lines = []
+    msg = ''.join((color,
+                   self.BOLD,
+                   main.__file__,
+                   ': ',
+                   record.outcome.name,
+                   self.RESET))
+
+    output_lines.append(msg)
+
+    if record.outcome in (test_record.Outcome.FAIL, test_record.Outcome.ERROR):
+      phases = record.phases
+      for phase in phases:
+        new_phase = True
+        phase_time = str((float(phase.end_time_millis)
+                          - float(phase.start_time_millis))/1000)
+        if record.outcome == test_record.Outcome.FAIL:
+          measured = phase.measured_values
+          measurements = phase.measurements
+          for key in measurements:
+            result = measurements[key]
+            if result.outcome == measurement.Outcome.FAIL:
+              if new_phase:
+                text = ''.join(('failed phase: ',
+                                phase.name,
+                                ' [time taken(s): ',
```

Comment:
prefer format strings for this sort of thing instead of `''.join()`, because it's easier to at-a-glance see what you're going for:
```
output_lines.append('failed phase: %s [ran for %.2f sec]' % (phase.name, phase_time_sec))
```